src/cpu/intel.rs

```diff
@@ -201,40 +202,67 @@ pub fn read_power_limit(file_path: String) -> f64 {
 }
 
 pub fn get_intel_cpu_cunter(results: &mut HashMap<String, f64>) {
+    // --- Read using Power Capping Framework (Linux only) ---
     // let zone = setup_rapl_data();
-    let start_time = Instant::now();
-    let mut prev_time = Instant::now();
-    let mut now = Instant::now();
-
+    // let start_time = Instant::now();
+    // let mut prev_time = Instant::now();
+    // let mut now = Instant::now();
     //    let mut data =
     //      calculate_power_metrics(zone.get(0).unwrap().to_owned(), now, start_time, prev_time);
 
     unsafe {
+        // --- Read using MSR ---
+
+        // TODO: these values are constant, read them once and store them
+        // The MSR only store integer values, but they represent floating point values.
+        // The MSR_RAPL_POWER_UNIT MSR contains the units for the RAPL MSRs for a specific intel chip.
+        // it contains three units, which represent the time, power, and energy increments in the RAPL MSRs.
         let core_energy_units: u64 = read_msr_on_core(MSR_RAPL_POWER_UNIT, 0).unwrap();
-        let energy_unit: u64 = (core_energy_units & INTEL_TIME_UNIT_MASK) >> 8;
+        
+        // First, we extract the individual units using the masks and offsets.
+        // Then we convert them to floating point values using the formula 0.5^x.
+        // See Section 14.9.1 of the Intel Architectures Software Developer's Manual (Vol 3B) for more information.
+
+        // Unused at the moment:
+        /*
+        let time_unit: u64 = (core_energy_units & INTEL_TIME_UNIT_MASK) >> INTEL_TIME_UNIT_OFFSET;
+        let time_unit_d = 0.5f64.powf(time_unit as f64);
+
+        let power_unit: u64 = (core_energy_units & INTEL_POWER_UNIT_MASK) >> INTEL_POWER_UNIT_OFFSET;
+        let power_unit_d = 0.5f64.powf(time_unit as f64);
+         */
+
+        let energy_unit: u64 = (core_energy_units & INTEL_ENGERY_UNIT_MASK) >> INTEL_ENGERY_UNIT_OFFSET;
         let energy_unit_d = 0.5f64.powf(energy_unit as f64);
 
+        // Read the RAPL MSRs
+        // PP0 = CPU cores energy consumption
         let pp0 = read_msr_on_core(INTEL_MSR_RAPL_PP0, 0).expect("failed to read PP0");
+        // PP1 = Integrated GPU energy consumption
         let pp1 = read_msr_on_core(INTEL_MSR_RAPL_PP1, 0).expect("failed to read PP1");
+        // PKG = CPU socket energy consumption
         let pkg = read_msr_on_core(MSR_RAPL_PKG_ENERGY_STAT, 0)
             .expect("failed to read RAPL_PKG_ENERGY_STAT");
+        // DRAM = Energy consumed by the DRAM for the chip's memory controller.
         let dram = read_msr_on_core(INTEL_MSR_RAPL_DRAM, 0).expect("failed to read DRAM");
-
+        
+        // convert the integer values to floating point values using the energy unit
+        // and store them in the results hashmap
         results.insert(
             format!("DRAM_ENERGY (J)"),
-            dram as f64 * energy_unit_d / UJ_TO_J_FACTOR,
```

Comment:
`UJ_TO_J_FACTOR` should only be used for the power capping framework. With a default value of `10000b = 16`, the energy unit is `(1/2)^16 ≈ 1.53e-6J`, which means the values are already in Joules and don't need to be converted from uJ to J.